homeassistant/helpers/device_registry.py

```diff
@@ -377,6 +484,13 @@ def async_get_or_create(
                     config_entry_id, connections, identifiers
                 )
             self.devices[device.id] = device
+            # If creating a new device, default to the config entry name
+            if (
+                device_info_type == "primary"
+                and (not name or name is UNDEFINED)
+                and config_entry
```

Comment:
In a follow-up we should fail on attempts to add devices to an unknown config entry

Reply:
Fixed in https://github.com/home-assistant/core/pull/98157